lib/project_types/script/layers/infrastructure/assemblyscript_dependency_manager.rb

```diff
@@ -22,19 +22,28 @@ def installed?
         end
 
         def install
+          check_node_version!
+
           output, status = @ctx.capture2e("npm", "install", "--no-audit", "--no-optional", "--loglevel error")
           raise Errors::DependencyInstallError, output unless status.success?
         end
 
         private
 
+        def check_node_version!
+          output, status = @ctx.capture2e("node", "--version")
+          raise Errors::DependencyInstallError, output unless status.success?
+
+          major, minor, * = output.split(".")
```

Comment:
Not critical, but I think we could have used `Semantic::Version` for this parsing.